tests/unit_tests/test_mutox.py

```diff
@@ -0,0 +1,105 @@
+    model = builder.build_model()
+
+    # Check if model layers are correctly initialized with shapes
+    assert isinstance(model, nn.Module), "Model should be an instance of nn.Module"
+    assert all(
+        isinstance(layer, nn.Sequential) for layer in model.model_all.children()
+    ), "All layers should be instances of nn.Sequential"
+
+    test_input = torch.zeros((5, input_size), device=device, dtype=dtype)
+    result = model(test_input)
+    assert result.shape == (5, 1), f"Expected output shape (5, 1), got {result.shape}"
+
+
+@pytest.mark.parametrize("input_size", [256, 512])
+def test_create_mutox_model(input_size):
+    """Test create_mutox_model function to confirm it creates a model with the specified config."""
+    config = MutoxConfig(input_size=input_size)
+    model = create_mutox_model(config, device=torch.device("cpu"))
+
+    # Check if the created model has the expected structure and behavior
+    test_input = torch.zeros((3, input_size))
+    result = model(test_input)
+    assert result.shape == (3, 1), f"Expected output shape (3, 1), got {result.shape}"
+    assert isinstance(model, nn.Module), "Model should be an instance of nn.Module"
+
+
+# Classifier tests
+
+
+def test_mutox_classifier_forward():
+    """Test that MutoxClassifier forward pass returns expected output shape."""
+    test_model = nn.Sequential(
+        nn.Linear(10, 5),
+        nn.ReLU(),
+        nn.Linear(5, 1),
+    )
+    model = MutoxClassifier(test_model)
+
+    test_input = torch.randn(3, 10)
+    output = model(test_input)
+    assert output.shape == (
+        3,
+        1,
+    ), f"Expected output shape (3, 1), but instead got {output.shape}"
+
+
+def test_mutox_config():
+    """Test that MutoxConfig stores the configuration for a model."""
+    config = MutoxConfig(input_size=512)
+    assert (
+        config.input_size == 512
+    ), f"Config input_size should be 512, but got {config.input_size}"
+
+
+#  Loader tests
+
+
+def test_convert_mutox_checkpoint():
+    """Test convert_mutox_checkpoint correctly filters keys in the checkpoint."""
+    # Create a mock checkpoint with both 'model_all.' prefixed keys and other keys
+    checkpoint = {
+        "model_all.layer1.weight": torch.tensor([1.0]),
+        "model_all.layer1.bias": torch.tensor([0.5]),
+        "non_model_key": torch.tensor([3.0]),
+    }
+    config = MutoxConfig(input_size=1024)
+    converted = convert_mutox_checkpoint(checkpoint, config)
+
+    # Verify only 'model_all.' keys are retained in the converted dictionary
+    assert "model" in converted, "Converted checkpoint should contain a 'model' key"
+    assert (
+        "model_all.layer1.weight" in converted["model"]
+    ), "Expected 'model_all.layer1.weight'"
+    assert (
+        "model_all.layer1.bias" in converted["model"]
+    ), "Expected 'model_all.layer1.bias'"
+    assert "non_model_key" not in converted["model"], "Unexpected 'non_model_key'"
```

Comment:
1. I think we also need a unit test for the `output_prob` argument. 
2. Please add an integration test that compares the outputs of the `sonar_mutox` classifier  for a few given texts with their expected values (e.g. the inputs and outputs that you already demonstrate in the readme). Just to make sure that we don't break the model in some unexpected way which is not detectable with unit tests that check only the output shape.

Reply:
I added integration tests for checking the output of both the model with linear layer output, and one for the sigmoid one. Now, the thing is that the output of the sigmoid, given the range that the model usually gives (high positive or negative numbers), is giving 0 or 1 for most input sentences. Should we add a "temperature" parameter of some sort? to try to make the values more informative

Reply:
No, I don't think temperature would make sense here, keep it as is.